lib/dice/src/handoff.rs

```diff
@@ -38,26 +60,42 @@ impl<'a> Handoff<'a> {
         self.0.ahbclkctrl2.modify(|_, w| w.usb1_ram().disable());
     }
 
-    pub fn store(&self, alias_data: &AliasData) -> usize {
-        // SAFETY: Dereferencing this raw pointer is necessary to write to the
-        // memory region used to handoff DICE artifacts to Hubris tasks. This
-        // pointer will references a valid memory region provided two
-        // conditions are met:
-        // 1) The associated memory region has been enabled / turned on if
-        // necessary. This happens in the constructor / 'turn_on' function.
-        // 2) The function call is made by code sufficintly privileged to
-        // access the memory region (e.g. stage0).
-        // If these conditions aren't met this access is still safe but a fault
-        // will occur.
-        let dst: &mut [u8] = unsafe {
-            core::slice::from_raw_parts_mut(
-                ALIAS_START as *mut u8,
-                AliasData::MAX_SIZE,
-            )
-        };
-
+    pub fn store<T>(&self, t: &T) -> usize
+    where
+        T: HandoffData + SerializedSize + Serialize,
+    {
+        let dst =
+            slice_from_parts(T::MEM_START, <T as SerializedSize>::MAX_SIZE);
         // TODO: error handling
-        hubpack::serialize(dst, alias_data).expect("serialize alias-handoff")
+        hubpack::serialize(dst, t).expect("handoff store")
+    }
+}
+
+pub trait HandoffData {
+    const EXPECTED_MAGIC: [u8; 16];
+    const MEM_START: usize;
+    const MEM_SIZE: usize;
+    const MAX_SIZE: usize;
+
+    fn get_magic(&self) -> [u8; 16];
+
+    fn from_mem() -> Option<Self>
+    where
+        Self: Sized,
+        for<'d> Self: Deserialize<'d>,
+    {
+        let src = slice_from_parts(Self::MEM_START, Self::MAX_SIZE);
```

Comment:
Nothing verifies that MAX_SIZE <= MEM_SIZE.  I'd feel better if this did something like:
`let src = slice_from_parts(Self::MEM_START, Self::MEM_SIZE)[:MAX_SIZE]`
or has a compile-time assert on MAX_SIZE <= MEM_SIZE.

Reply:
I've updated the PR with static assertions in each implementer of HandoffData to do this check.